sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,236 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImportEdifact(models.TransientModel):
+    _name = "sale.order.import"
+    _inherit = ["sale.order.import", "base.edifact"]
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    state = fields.Selection(
+        selection_add=[("tech", "Technical")], ondelete={"tech": "set default"}
+    )
+    parsed_json = fields.Text()
+
+    @property
+    def edifact_ok(self):
+        conf_ext = self.env.context.get("file_ext", "txt,d96a")
+        extensions = conf_ext.split(",")
+        ok = False
+        if self.order_filename:
+            path, ext = os.path.splitext(self.order_filename)
+            ok = ext and ext[1:] in extensions
+            if not ok and self.order_file:
+                ok = b64decode(self.order_file[:4]) == b"UNB"
+        return ok
+
+    @api.onchange("order_file")
+    def order_file_change(self):
+        if self.edifact_ok:
+            self.doc_type = self.env.context.get("doc_type", "rfq")
+            self.price_source = "order"
+        else:
+            res = super(SaleOrderImportEdifact, self).order_file_change()
+            if isinstance(res, dict):
+                return res
+
+    def parse_order_button(self):
```

Comment:
hmm this seems to be an additional method for the btn that is not defined by the base module..

Reply:
This button was added by the original author of the module in v14. So do you want me to remove this button?

Reply:
It's just for debugging (groups="base.group_system,base.group_no_one"). To take a quick look at the pydifact structure ('edifact' field) and the parsed object ('order') from the original document before importing the document.

It was very useful during development and I left it to detect changes in the future